src/sagemaker/workflow/emr_step.py

```diff
@@ -155,6 +163,7 @@ def __init__(
         depends_on: Optional[List[Union[str, Step, StepCollection]]] = None,
         cache_config: CacheConfig = None,
         cluster_config: Dict[str, Any] = None,
+        execution_role_arn: str = None,
```

Comment:
`step_config` is mapped to [StepConfig](https://docs.aws.amazon.com/emr/latest/APIReference/API_StepConfig.html) in EMR's [AddJobFlowSteps](https://docs.aws.amazon.com/emr/latest/APIReference/API_AddJobFlowSteps.html#EMR-AddJobFlowSteps-request-Steps), it's not a single-step configuration but applies to all steps to run in a single AddJobFlowSteps (though Pipelines does only run one job step per EMR step).

Reply:
@qidewenwhen are you suggesting using kwargs intead?

Reply:
I see, then `step_config` is not a good place to go, in terms of `execution_role_arn`.

I'm just hoping to raise discussion and caution on this public facing interface change as it's not revertible.
And I do have a concern that if EMR side keeps adding new args, our EMRStep interface will keep expanding and eventually it may be hard to use, see example of the obsoleted `RegisterModel` interface which contains ~30 args: https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/workflow/step_collections.py#L57-L94

Reply:
So **kwargs seems to be able to resolve my concern above but usually it may not be a good practice to include **kwargs.
Thus I'd suggest to have thorough consideration on this step interface and have more eyes on it.

Reply:
Synced up offline: there's no better approach to avoid the call out issue. We will introduce a `step_arg` parameter to group all EMRStep argument related parameters in the future when the step interface grows too big.